### PR TITLE
Azure Wodle - Fix InvalidRange error

### PR DESCRIPTION
**Pull request by RamosFe**
|Related issue|
|---|
|[300](https://github.com/wazuh/support/issues/300)|

## Description

Added new validation for the Azure Wodle by checking the blob size before requesting the blob's content from Azure to ensure we don't have the InvalidRange error.

### CODE REVIEW

**Comment by GGP1** [comment]
We need to make sure all checks pass to approve the review.

File: wodles/azure/azure-logs.py [hunk 1]
```diff
@@ -718,6 +718,10 @@ def get_blobs(
                     min_datetime <= last_modified <= max_datetime)):
                 continue
 
+            # Skip if the blob is empty
+            if blob.properties.content_length == 0:
+                continue
+
```
[the remark this answers is not in the document]
Comment (RamosFe) [reply wodles/azure/azure-logs.py]: Added change in commit 82be9b094b4ee33a8d6fbd0b50f9b05252b7e0d2

**Review by GGP1 — APPROVED** [review]
Tested the azure-logs script with an empty blob and unit tests, both worked as expected. LGTM

## Empty blob test

```console
(venv2) gasti@pop-os:~/work/wazuh/wodles$ python azure/azure-logs.py --storage --storage_auth_path /var/ossec/etc/azure_credentials --container emptyblob -d 2
2023/05/09 02:43:58 azure: INFO: Checking database integrity
2023/05/09 02:43:58 azure: INFO: Database integrity check finished
2023/05/09 02:43:58 azure: INFO: Azure Storage starting.
2023/05/09 02:43:58 azure: INFO: Storage: Authenticating.
2023/05/09 02:43:58 azure: DEBUG: String_to_sign=GET
x-ms-client-request-id:xxx
x-ms-date:Tue, 09 May 2023 17:43:58 GMT
x-ms-version:2019-02-02
/frameworkteststorage/emptyblob
restype:container
2023/05/09 02:43:58 azure: INFO: Client-Request-ID=xxx Outgoing request: Method=GET, Path=/emptyblob, Query={'snapshot': None, 'timeout': None, 'restype': 'container'}, Headers={'x-ms-version': '2019-02-02', 'User-Agent': 'Azure-Storage/2.1.0-2.1.0 (Python CPython 3.9.16; Linux 6.2.6-76060206-generic)', 'x-ms-client-request-id': 'xxx', 'x-ms-date': 'Tue, 09 May 2023 17:43:58 GMT', 'Authorization': 'REDACTED'}.
2023/05/09 02:43:58 azure: INFO: Client-Request-ID=xxx Receiving Response: Server-Timestamp=Tue, 09 May 2023 17:43:57 GMT, Server-Request-ID=xxx, HTTP Status Code=200, Message=OK, Headers={'content-length': '0', 'last-modified': 'Wed, 19 Apr 2023 16:51:38 GMT', 'etag': '"xxx"', 'server': 'Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0', 'x-ms-request-id': 'xxx', 'x-ms-client-request-id': 'xxx', 'x-ms-version': 'xxx', 'x-ms-lease-status': 'unlocked', 'x-ms-lease-state': 'available', 'x-ms-has-immutability-policy': 'false', 'x-ms-has-legal-hold': 'false', 'x-ms-default-encryption-scope': '$account-encryption-key', 'x-ms-deny-encryption-scope-override': 'false', 'date': 'Tue, 09 May 2023 17:43:57 GMT'}.
2023/05/09 02:43:58 azure: INFO: Storage: Authenticated.
2023/05/09 02:43:58 azure: INFO: Storage: Getting blobs.
2023/05/09 02:43:58 azure: DEBUG: String_to_sign=GET
x-ms-client-request-id:xxx
x-ms-date:Tue, 09 May 2023 17:43:58 GMT
x-ms-version:xxx
/xxx/emptyblob
comp:list
restype:container
2023/05/09 02:43:58 azure: INFO: Client-Request-ID=xxx Outgoing request: Method=GET, Path=/emptyblob, Query={'restype': 'container', 'comp': 'list', 'prefix': None, 'delimiter': None, 'marker': None, 'maxresults': None, 'include': None, 'timeout': None}, Headers={'x-ms-version': '2019-02-02', 'User-Agent': 'Azure-Storage/2.1.0-2.1.0 (Python CPython 3.9.16; Linux 6.2.6-76060206-generic)', 'x-ms-client-request-id': 'xxx', 'x-ms-date': 'Tue, 09 May 2023 17:43:58 GMT', 'Authorization': 'REDACTED'}.
2023/05/09 02:43:58 azure: INFO: Client-Request-ID=xxx Receiving Response: Server-Timestamp=Tue, 09 May 2023 17:43:57 GMT, Server-Request-ID=xxx, HTTP Status Code=200, Message=OK, Headers={'transfer-encoding': 'chunked', 'content-type': 'application/xml', 'server': 'Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0', 'x-ms-request-id': 'xxx', 'x-ms-client-request-id': 'xxx', 'x-ms-version': 'xxx', 'date': 'Tue, 09 May 2023 17:43:57 GMT'}.
2023/05/09 02:43:58 azure: INFO: Storage: The search starts from the date: 2023-05-09 00:00:00+00:00 for blobs in container: 'emptyblob' and prefix: '/'
2023/05/09 02:43:58 azure: INFO: Storage: End
```

### Azure unit tests

```console
venv2) gasti@pop-os:~/work/wazuh/wodles$ python -m pytest azure/tests/test_azure.py
===================================== test session starts ======================================
platform linux -- Python 3.9.16, pytest-5.4.3, py-1.11.0, pluggy-0.13.1
rootdir: /home/gasti/work/wazuh/wodles
plugins: aiohttp-0.3.0, tavern-1.2.2, trio-0.7.0, html-2.1.1, asyncio-0.15.1, metadata-2.0.4
collected 116 items                                                                            

azure/tests/test_azure.py .............................................................. [ 53%]
......................................................                                   [100%]

===================================== 116 passed in 0.44s ======================================

```

**Comment by davidjiglesias** [comment]
The cmocka fail is known as is already fixed for next PRs.

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**